llarp/iwp/message_buffer.hpp

```diff
@@ -49,6 +50,13 @@ namespace llarp
       llarp_time_t m_LastFlush = 0s;
       ShortHash m_Digest;
       llarp_time_t m_StartedAt = 0s;
+      uint16_t m_ResendPriority;
+
+      bool
+      operator<(const OutboundMessage& msg) const
+      {
+        return msg.m_ResendPriority < m_ResendPriority;
+      }
```

Comment:
Rather than defining `operator <` in a way that seems wrong at first glance, it might be better to define this `operator<` the "natural" way and then use a templated typedef in a util header somewhere to define a lower-is-better priority queue:
```C++
template <typename T, typename Container = std::vector<T>>
using ascending_priority_queue = std::priority_queue<T, Container, std::greater<typename Container::value_type>>;
```
which you can use as `negative_priority_queue<MyType> stuff{};` to give you a smaller-comes-first priority queue.